pulsar/consumer_partition.go

```diff
@@ -1760,6 +1770,21 @@ func (pc *partitionConsumer) markScaleIfNeed() {
 	}
 }
 
+func (pc *partitionConsumer) shrinkReceiverQueueSize() {
+	if !pc.options.autoReceiverQueueSize {
+		return
+	}
+
+	oldSize := pc.currentQueueSize.Load()
```

Comment:
We still need `receiverQueueShrinking ` to protect can only be shrinking once at a time.

Reply:
oh, I see. Already protected in `memory_limit_controller`.